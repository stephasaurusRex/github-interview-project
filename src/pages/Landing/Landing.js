import React, { Component } from "react";
import Card from "../../components/Cards/Card";
import {initiateGitHub, getRepos, getTopContributor} from "../../services/GitHubService";

class Landing extends Component {

  state = {
    repos: [],
  }

  async componentDidMount() {
    this.mounted = true;
    try {
      initiateGitHub();
      getRepos().then((body) => {
        if(this.mounted) {
          this.setState({
            repos: body.data.items,
          });
        }
        body.data.items.forEach((repo, idx) => {
          getTopContributor(repo)
            .then((newRepo) => {
              body.data.items[idx] = newRepo;
              if(this.mounted) {
                this.setState({repos: body.data.items})
              }
            })
            .catch(() => {
              repo.topContributor = { login: 'Error Occurred' };
              body.data.items[idx] = repo;
              if(this.mounted) {
                this.setState({repos: body.data.items})
              }
            })
        });

      });
    } catch (error) {
      console.log(error);
    }
  }

  componentWillUnmount() {
    this.mounted = false;
  }

  render() {
    return (
      <div className="grid-container">
        {
        this.state.repos && this.state.repos.length > 0 && this.state.repos.map((repo, idx) =>
          (
            <Card
              key={repo.id}
              rank={idx + 1}
              repo={repo}>
            </Card>
          )
        )
      }
        </div>
    )
  }
}

export default Landing;
