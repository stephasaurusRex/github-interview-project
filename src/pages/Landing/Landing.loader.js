// split the bundle on a page level to limit the size of the initial load
// https://github.com/jamiebuilds/react-loadable

import React from "react";
import Loadable from "react-loadable";

/* component rendered while the landing page is loaded */

export default Loadable({
  loader: () => import("./Landing"),
  loading: () => <LoadingLanding />,
});
