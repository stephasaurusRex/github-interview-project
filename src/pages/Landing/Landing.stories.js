import { withInfo } from "@storybook/addon-info";
import { storiesOf } from "@storybook/react";
import React from "react";

import Landing from "./Landing";

storiesOf("landing page", module)
  .add("basic usage", withInfo(`The landing page`)(() =>
        <Landing />
  ));
