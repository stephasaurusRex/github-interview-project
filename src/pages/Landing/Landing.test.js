import "jest-styled-components";

import React from "react";
import TestRenderer from "react-test-renderer";

import Landing from "./Landing";

describe("Landing", () => {

  it("renders without props", () => {
    const testRenderer = TestRenderer.create(
      <Landing/>
    );
    expect(testRenderer).toMatchSnapshot();
    testRenderer.unmount();
  });

});
