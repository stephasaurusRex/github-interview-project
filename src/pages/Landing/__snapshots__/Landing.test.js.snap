// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing renders without props 1`] = `
<div
  className="grid-container"
/>
`;
